python/dolfin/function/constant.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 Michal Habera, Chris Richardson
+#
+# This file is part of DOLFIN (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+
+import numpy
+
+import ufl
+import dolfin.cpp
+
+
+class Constant(ufl.Constant):
+    def __init__(self, domain, value):
+        np_value = numpy.asarray(value)
+        super().__init__(domain, np_value.shape)
+        self._cpp_object = dolfin.cpp.function.Constant(np_value.flatten())
+
+    @property
+    def value(self):
+        val = self._cpp_object.value
```

Comment:
Would be cleaner to handle type in the pybind11 layer.

Reply:
I put it here so it can return different types, e.g. a float, or a n-dimensional numpy array. Not easy to do in C++ with `variant` etc. I'm not sure if it is even possible for n>2 (for example Cijkl tensor) - does pybind work with `Eigen::Tensor`?

Reply:
No need to involve Eigen. Suggestion is to wrap the C++ data as a NumPy array in the pybdind11 layer.

Reply:
It's nice to be able to return a `float` when there is just one value.